### PR TITLE
Added hazelcast-azure to hazelcast-all

### PR DESCRIPTION
**Pull request by alparslanavci**
.. and updated the configs with the latest changes in Hazelcast Azure plugin.

### CODE REVIEW

**Comment by hz-highfive** [comment]
The job `Hazelcast-pr-builder` of your PR failed ([log](
https://j-artifacts.s3.amazonaws.com/Hazelcast-pr-builder/625/build_log.txt), [artifacts](https://j-artifacts.s3.amazonaws.com/Hazelcast-pr-builder/625/Hazelcast-pr-builder-625.zip)). 
Through arcane magic we have determined that the following fragments from the build log may contain information about the problem.
<details>
<summary>Click to expand the log file</summary>
<pre>
--------------------------
-------TEST FAILURE-------
--------------------------
[INFO] Results:
[INFO] 
[ERROR] Failures: 
[ERROR]   ClientCacheEventJournalBasicTest>AbstractEventJournalBasicTest.allowReadingWithFutureSeq:512->HazelcastTestSupport.assertOpenEventually:1108->HazelcastTestSupport.assertOpenEventually:1123 CountDownLatch failed to complete within 30 seconds, count left: 1
[INFO] 
[ERROR] Tests run: 27729, Failures: 1, Errors: 0, Skipped: 1001
[INFO] 

[ERROR] There are test failures.
</details>
</pre>

**Comment by alparslanavci** [comment]
verify

**Comment by alparslanavci** [comment]
run-lab-run

**Comment by hz-highfive** [comment]
The job `Hazelcast-pr-builder` of your PR failed ([log](
https://j-artifacts.s3.amazonaws.com/Hazelcast-pr-builder/628/build_log.txt), [artifacts](https://j-artifacts.s3.amazonaws.com/Hazelcast-pr-builder/628/Hazelcast-pr-builder-628.zip)). 
Through arcane magic we have determined that the following fragments from the build log may contain information about the problem.
<details>
<summary>Click to expand the log file</summary>
<pre>
--------------------------
-------TEST FAILURE-------
--------------------------
[INFO] Results:
[INFO] 
[ERROR] Failures: 
[ERROR]   ClientMapEventJournalBasicTest>AbstractEventJournalBasicTest.allowReadingWithFutureSeq:512->HazelcastTestSupport.assertOpenEventually:1108->HazelcastTestSupport.assertOpenEventually:1123 CountDownLatch failed to complete within 30 seconds, count left: 1
[INFO] 
[ERROR] Tests run: 27729, Failures: 1, Errors: 0, Skipped: 1001
[INFO] 

[ERROR] There are test failures.
</details>
</pre>

**Comment by alparslanavci** [comment]
run-lab-run

**Comment by alparslanavci** [comment]
run-lab-run

**Comment by mesutcelik** [comment]
@alparslanavci Do you have a backport PR for 4.0.z?

**Comment by alparslanavci** [comment]
> @alparslanavci Do you have a backport PR for 4.0.z?

Not yet, but I can add it after merging this.

**Comment by jerrinot** [comment]
is it wise to backport this?

**Comment by mesutcelik** [comment]
I see that can help increase 4.0 adoption but if you see a risk then I am ok. It is not a hard requirement.

For instance, this blog post showcased hazelcast 4.0 and hazelcast-azure plugin but without hazelcast-all
https://dev.to/azure/java-distributed-caching-in-the-cloud-with-the-new-azure-discovery-plugin-for-hazelcast-bme
https://github.com/jdubois/jhipster-hazelcast-azure/blob/master/pom.xml#L148

**Comment by jerrinot** [comment]
right, makes sense. let's backport it. please make sure it's inlcluded in release notes.

File: hazelcast/src/main/resources/hazelcast-default.xml [hunk 1]
```diff
@@ -69,12 +69,6 @@
                 <zones>us-east1-b,us-east1-c</zones>
             </gcp>
             <azure enabled="false">
-                <client-id>CLIENT_ID</client-id>
-                <client-secret>CLIENT_SECRET</client-secret>
-                <tenant-id>TENANT_ID</tenant-id>
-                <subscription-id>SUB_ID</subscription-id>
-                <cluster-id>HZLCAST001</cluster-id>
-                <group-name>RESOURCE-GROUP-NAME</group-name>
```
Comment (blazember) [inline hazelcast/src/main/resources/hazelcast-default.xml:77]: Are these removed on purpose? The rest of the discovery plugins have default values set explicitly.

Comment (alparslanavci) [reply hazelcast/src/main/resources/hazelcast-default.xml:77]: Yes, we don't need any properties by default on member config if the user sets up the identites (See more detail here: https://github.com/hazelcast/hazelcast-azure#basic-configuration). For full config, I added all available properties. Please see `hazelcast-full-example.xml`.

**Pull request merged**